src/xfieldhistory.cpp

```diff
@@ -20,6 +22,7 @@ bool CXFieldHistoryMap::IsNew(TAPYRUS_XFIELDTYPES type, const XFieldChange& xFie
 }
 
 void CXFieldHistoryMap::Add(TAPYRUS_XFIELDTYPES type, const XFieldChange& xFieldChange) {
+    LOCK(xfieldHistoryMutex);
```

Comment:
Why is this modification, including xfieldhistory.h, necessary for CMake?

Reply:
When I was running `cmake ci ` `ctest` command it ran running `test_tapyrus` with multiple threads. There was a risk of the global data being corrupted when `xfieldhistory` run in parallel with other tests. Suspecting this as the cause of test failures I added this code to make the access synchronous.
With `ctest` I couldn't find how to forward arguments to `test_tapyrus`. So I changed the CI to use `test_tayrus` command directly with all the arguments.